src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+     * @param dbSNPInput input for variants from a known set from DbSNP. The annotation engine will mark variants overlapping anything in this set using {@link htsjdk.variant.vcf.VCFConstants#DBSNP_KEY}.
+     */
+    public static VariantAnnotatorEngine ofSelectedMinusExcluded(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude, final FeatureInput<VariantContext> dbSNPInput) {
+        Utils.nonNull(annotationGroupsToUse, "annotationGroupsToUse is null");
+        Utils.nonNull(annotationsToUse, "annotationsToUse is null");
+        Utils.nonNull(annotationsToExclude, "annotationsToExclude is null");
+        return new VariantAnnotatorEngine(AnnotationManager.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude), dbSNPInput);
+    }
+
+    private VariantOverlapAnnotator initializeDBs(final FeatureInput<VariantContext> dbSNPInput) {
+        final Map<FeatureInput<VariantContext>, String> overlaps = new LinkedHashMap<>();
+        if ( dbSNPInput != null ) {
+            overlaps.put(dbSNPInput, VCFConstants.DBSNP_KEY); // add overlap detection with DBSNP by default
+        }
+
+        return new VariantOverlapAnnotator(dbSNPInput, overlaps);
+    }
+
+    /**
+     * Returns the list of genotype annotations that will be applied.
+     * Note: The returned list is unmodifiable.
+     */
+    public List<GenotypeAnnotation> getGenotypeAnnotations() {
+        return Collections.unmodifiableList(genotypeAnnotations);
+    }
+
+    /**
+     * Returns the list of info annotations that will be applied.
+     * Note: The returned list is unmodifiable.
+     */
+    public List<InfoFieldAnnotation> getInfoAnnotations() {
+        return Collections.unmodifiableList(infoAnnotations);
+    }
+
+    /**
+     * Returns the set of descriptions to be added to the VCFHeader line (for all annotations in this engine).
+     */
+    public Set<VCFHeaderLine> getVCFAnnotationDescriptions() {
+        final Set<VCFHeaderLine> descriptions = new HashSet<>();
+
+        for ( final InfoFieldAnnotation annotation : infoAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final GenotypeAnnotation annotation : genotypeAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final String db : variantOverlapAnnotator.getOverlapNames() ) {
+            if ( VCFStandardHeaderLines.getInfoLine(db, false) != null ) {
+                descriptions.add(VCFStandardHeaderLines.getInfoLine(db));
+            } else {
+                descriptions.add(new VCFInfoHeaderLine(db, 0, VCFHeaderLineType.Flag, db + " Membership"));
+            }
+        }
+
+        return descriptions;
+    }
+
+    /**
+     * Annotates the given variant context - adds all annotations that satisfy the predicate.
+     * @param features context containing the features that overlap the given variant
```

Comment:
same convention - unless we explicitly say that X can be null then X cannot be null.